### ISSUE TITLE
known_deprecations issues with multiple names

### ISSUE THREAD

**Issue by RomeshA** [sciris/sciris#502]
Currently `known_deprecations` has

```
    known = {
        "No module named 'pandas.core.indexes.numeric'":
            dict(module = 'pandas',
                 pickled_version = '<2.0',
                 current_version = '>=2.0',
                 fix = {'pandas.core.indexes.numeric.Int64Index':'pandas.core.indexes.api.Index'},
        ),
    }
```

but `Int64Index` is not the only affected class. I'm encountering an error with a dataframe that has a `Float64Index` index, but the error still presents as `"No module named 'pandas.core.indexes.numeric'"` so when the known deprecation is added to `remapping`, loading the pickle still fails because the fix only applies to `Int64Index`

**Comment by RomeshA** [comment]
Worth noting in this case that `pandas.read_pickle` seems to load the file without needing any workarounds - that function is meant to provide backwards-compatible loading of dataframes. It could be worth considering providing that as a fallback option if loading with `pickle` fails with a Pandas error (presumably if the original pickle fell back to `dill` then loading with `read_pickle()` wouldn't work anyway)?

**Comment by cliffckerr** [comment]
See also how pandas does it:
https://github.com/pandas-dev/pandas/blob/main/pandas/tests/io/generate_legacy_storage_files.py

**Comment by cliffckerr** [comment]
Closed as part of #503
